_posts/2023-06-23-run-and-debug-rust-verification-harnesses-natively-on-vscode.md

```diff
@@ -0,0 +1,174 @@
+This empowers our users to find bugs and verify their code quicker.
+You can install the extension from the [webpage](https://marketplace.visualstudio.com/items?itemName=model-checking.kani-vscode-extension) on VS Code marketplace or by searching for `Kani` in the extensions tab of your VS Code instance.
+
+## Using the VS Code extension
+
+#### How to verify a harness using the Kani VS Code extension
+
+With the extension, running Kani on a harness to verify it, is as simple as clicking a button.
+The following sections walk you through the first few actions you’ll need.
+We’ll walk you through using the extension’s core features to debug and finally verify the rectangle example mentioned above.
+
+#### View Kani harnesses
+
+As soon as your rust package is opened using the Kani extension in a VS Code instance, you should see the Kani proofs loaded as regular unit tests in the Testing Panel on the [primary side bar](https://code.visualstudio.com/api/ux-guidelines/sidebars#primary-sidebar) of VS Code.
+This is how the testing page looks like when you click on the panel.
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/view-kani-harnesses.png" alt="view-kani-harnesses" />
+
+#### Run Kani harnesses
+
+You can then run your harnesses using the tree view by clicking the play button beside the harness that was automatically picked up by the Kani Extension. Once you run the harness using the extension, you are shown a failure banner if verification failed (or a green check mark if it succeeded).
+
+In our example, as with the command line, we can see through visual markers such as the failure banner pop-up and the red failure marker, that verification failed.
+
+You are then presented with two options:
+
+1. [Generate the report for the harness](https://github.com/model-checking/kani-vscode-extension/blob/main/docs/user-guide.md#view-trace-report)
+2. [Run concrete playback to generate unit tests](https://github.com/model-checking/kani-vscode-extension/blob/main/docs/user-guide.md#use-concrete-playback-to-debug-a-kani-harness).
+
+<img src="{{site.baseurl | prepend: site.url}}/assets/images/vs-code-images/run-kani-harness.gif" style="box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);" alt="run-kani-harness" />
+
+Kani can help you generate unit tests containing the counter-example (or values for which the assertion fails).
+Each unit test provides inputs that will either trigger a property failure or satisfy a cover statement.
+This feature, called [concrete playback](https://model-checking.github.io/kani-verifier-blog/2022/09/22/internship-projects-2022-concrete-playback.html), allows you to generate unit tests that call a function with the exact arguments that caused the assertion violation, and the VSCode extension makes using concrete playback easy.
+You can read more about concrete playback in our [documentation](https://model-checking.github.io/kani/debugging-verification-failures.html).
```

Comment:
Not sure why we're talking about concrete playback here, with this level of detail.

Reply:
Edited this section to be shorter